Enhancement for the following GUI wizard installation option.
1) If Xen, leave 'Architecture options' expanded so users know PV
is the default
Index: virt-manager-2.2.1/virtManager/createvm.py
===================================================================
--- virt-manager-2.2.1.orig/virtManager/createvm.py
+++ virt-manager-2.2.1/virtManager/createvm.py
@@ -404,8 +404,20 @@ class vmmCreateVM(vmmGObjectUI):
         self.widget("method-local").set_active(True)
         self.widget("create-conn").set_active(-1)
         activeconn = self._populate_conn_list(urihint)
-        self.widget("arch-expander").set_expanded(False)
         self.widget("vz-virt-type-hvm").set_active(True)
+        # For Xen have the expander open so users can see PV is the default
+        if activeconn and activeconn.is_xen():
+            self.widget("arch-expander").set_expanded(True)
+        else:
+            self.widget("arch-expander").set_expanded(False)
+
+        # Default to Network install if host was installed that way
+        host_instsrc = getHostInstallSource()
+        if host_instsrc is not None and \
+           (host_instsrc.startswith('ftp:') or host_instsrc.startswith('http:') or \
+            host_instsrc.startswith('smb:') or host_instsrc.startswith('nfs:')):
+            self.widget("method-local").set_active(False)
+            self.widget("method-tree").set_active(True)
 
         if self._set_conn(activeconn) is False:
             return False
